signer/src/transaction_coordinator.rs

```diff
@@ -284,19 +285,29 @@ where
         tracing::debug!("We are the coordinator, we may need to coordinate DKG");
         // If Self::get_signer_set_and_aggregate_key did not return an
         // aggregate key, then we know that we have not run DKG yet. Since
-        // we are the signer, we should coordinate DKG.
+        // we are the coordinator, we should coordinate DKG.
         let aggregate_key = match maybe_aggregate_key {
             Some(key) => key,
             // This function returns the new DKG aggregate key.
-            None => self.coordinate_dkg(&bitcoin_chain_tip).await?,
+            None => {
+                self.coordinate_dkg(&bitcoin_chain_tip).await?;
+                let (maybe_aggregate_key, _) = self
+                    .get_signer_set_and_aggregate_key(&bitcoin_chain_tip)
+                    .await?;
+                maybe_aggregate_key.ok_or(Error::NoDkgShares)?
```

Comment:
Ahh there is a race condition with the transaction signer. Well, I don't think that we should bail here, we could still deploy the smart contracts.

Reply:
Yeah this is actually not solving the underlying issue, reverting the change.

Reply:
Nvm, we may need this, added a comment about it (and removed the bailing).